syntax = "proto3";

package simulator;

message AudioStream {
	bytes data = 1;
}

message HistoryResponse {
	map<string, string> history = 1;
}

message HistoryRequest {}

message PromptRequest {
    string prompt = 1;
}

message PromptResponse {
    string status = 1;
}

service SimulatorService {
	rpc StreamAudio(stream AudioStream) returns (stream AudioStream);
	rpc GetConversationHistory(HistoryRequest) returns (HistoryResponse);
    rpc ReceivePrompt(PromptRequest) returns (PromptResponse);
}
